libsel4benchsupport/include/sel4benchsupport/signal.h

```diff
@@ -13,7 +13,14 @@
 
 typedef struct signal_results {
```

Comment:
This would be improved by seperating the two structures so that it's clear we are either defining early reuslts or not. using a shared struct where some fields are present vs not based on a config setting could be confusing to workout for readers without context.

Reply:
The caller of the "signal benchmark" function deals with a single structure "signal_result_t" (allocates memory for it). Now the both methods co-exist, I don't see benefits to have two separate "storages". I re-organized the structure by grouping data related to different methods and I added comments.